spec/oas3.fetched/components/schemas/MetalGateway.yaml

```diff
@@ -29,4 +29,7 @@ properties:
     type: string
   virtual_network:
     $ref: './VirtualNetwork.yaml'
+  vrf:
```

Comment:
Non-VRF and VRF Metal Gateways appear to have the same properties, with the sole exception that VRF Metal Gateways have a `vrf` property.

As such, there isn't a field we can put under `required` in order to identify non-VRF gateways.  Ideally the API response would include a field (e.g., `type: vrf` and `type: non-vrf`) that we could use to differentiate the gateway schemas more directly, but that doesn't exist today.  We also can't immediately introduce such a field, because `metal-java` addressed this issue by setting `additionalProperties: false` on the MetalGateway schema, which means that if API responses start including additional properties for non-VRF Metal Gateways, all existing SDKs will throw errors about invalid responses.

For VRF Metal Gateways, we would need to update the spec to mark `vrf` as `required`, but that won't work until we do some work to enable required field validation: #132

Reply:
I was trying to remember how I got this working without `required` support...  The `MetalGateway` and `VrfMetalGateway` schemas can be disambiguated by leveraging the `type` field on IP reservations: a VRF IP reservation has one valid type, `vrf`, and a non-VRF IP reservation has multiple value types, none of which are `vrf`.  Since the type for each IP reservation schema is defined as an enum, `metal-go` is now capable of using that field to determine which schema to use; since a `MetalGateway` always has a non-VRF IP reservation and a `VrfMetalGateway` always has a VRF IP reservation, we can `include=ip_reservation` to help `metal-go` figure out the right gateway schema: https://github.com/equinix-labs/metal-go/pull/127/files#diff-49d8c1d06b64f8f0e02a49dd8b5f8efc1881471d3324bfbcf62e9f4403e311e0R86-R87

Reply:
As described in the comments above, we can leverage type differentiation for child schema's (the IP reservation attached to the gateway) in order to determine the type of the gateway.  As such, this schema workaround isn't needed, but I'm going to start a conversation with the API team about introducing a `type` property on gateways similar to what was added on virtual circuits.